demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "custom_kernels.hpp"
+#include <opencv2/imgproc.hpp>
+#include <list>
+
+const float BOUNDING_BOX_THRESHOLD = 0.4f;
+const int ACTION_IMAGE_SCALE = 256;
+
+namespace {
+cv::Rect convert_to_central_roi(const cv::Rect& person_roi,
+                                const cv::Size& in_size,
+                                const size_t scale) {
+    const int roi_height = person_roi.height;
+    const int roi_width = person_roi.width;
+
+    const int src_roi_center_x = int(0.5 * (person_roi.tl().x + person_roi.br().x));
+    const int src_roi_center_y = int(0.5 * (person_roi.tl().y + person_roi.br().y));
+
+    const float height_scale = float(in_size.height) / float(scale);
+    const float width_scale = float(in_size.width) / float(scale);
+
+    CV_DbgAssert(height_scale < 1.0f);
+    CV_DbgAssert(width_scale < 1.0f);
+
+    const int min_roi_size = std::min(roi_height, roi_width);
+    const int trg_roi_height = int(height_scale * min_roi_size);
+    const int trg_roi_width  = int(width_scale * min_roi_size);
+
+    return cv::Rect(int(src_roi_center_x - 0.5 * trg_roi_width),
+                    int(src_roi_center_y - 0.5 * trg_roi_height),
+                    trg_roi_width,
+                    trg_roi_height);
+}
+} // anonymous namespace
+
+GAPI_OCV_KERNEL(OCVGetFastFrame, custom::GetFastFrame) {
+    static void run(const std::vector<cv::Mat>& batch,
+                    const cv::Size& image_size,
+                          cv::Mat& frame) {
+        const uint8_t* ptrI = batch[batch.size() - 2].ptr<uint8_t>();
+        std::copy(ptrI, ptrI + image_size.area() * 3, frame.ptr<uint8_t>());
```

Comment:
Because the frame has framerate of source video. We get the frame from cyclic buffer. Buffer contains batch for ASL-net and fast frame. Batch framerate is constant (15 FPS).